### PR TITLE
Add LRange commad

### PR DESCRIPTION
**Pull request by Reid00**
Related issue: https://github.com/flower-corp/rosedb/issues/149

- refactor listSequence to apply it in LRange function
- add listSequence test cases
- Add LRange function

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -44,6 +44,9 @@ var (
 	// ErrWrongIndex index is out of range
 	ErrWrongIndex = errors.New("index is out of range")
 
+	// ErrIndexStartLagerThanEnd start index lager than end index after conversion
+	ErrIndexStartLagerThanEnd = errors.New("start physical seq lager than end physical seq")
```
Comment (gozeloglu) [inline db.go:48]: Typo in many places: `Lager`

Comment (Reid00) [reply db.go]: fixed

**Review by gozeloglu — COMMENTED** [review]
Can you add `LRange` command to the CLI part?

**Comment by Reid00** [comment]
I didn't see any code about CLI part currently, I will check this part tomorrow and add it.

**Comment by gozeloglu** [comment]
> I didn't see any code about CLI part currently, I will check this part tomorrow and add it.

Here: https://github.com/flower-corp/rosedb/blob/main/cmd/cli.go

**Comment by Reid00** [comment]
add CLI part for LRange and LSet, pls take a look.

**Review by gozeloglu — APPROVED** [review]
```
   __   _____________  ___
  / /  / ___/_  __/  |/  /
 / /__/ (_ / / / / /|_/ / 
/____/\___/ /_/ /_/  /_/  
```

File: db.go [hunk 2]
```diff
@@ -44,6 +44,9 @@ var (
 	// ErrWrongIndex index is out of range
 	ErrWrongIndex = errors.New("index is out of range")
 
+	// ErrIndexStartLargerThanEnd start index larger than end index after conversion
+	ErrIndexStartLargerThanEnd = errors.New("start physical seq larger than end physical seq")
```
Comment (roseduan) [inline db.go:48]: this err is redundant, return ErrWrongIndex is the same?

File: list.go [hunk 3]
```diff
@@ -193,6 +198,71 @@ func (db *RoseDB) LSet(key []byte, index int, value []byte) error {
...
+	}
+	endSeq, err = db.listSequence(headSeq, tailSeq, end)
+	if err != nil {
+		return nil, err
+	}
+	// normalize startSeq
+	if startSeq <= headSeq {
+		startSeq = headSeq + 1
+	}
+	if startSeq >= tailSeq {
```
Comment (roseduan) [inline list.go:238]: You can combine all conditions into one, that looks like more clear.
if startSeq >= tailSeq || endSeq <= headSeq || startSeq > endSeq {
    return nil, ErrWrongIndex
}

**Comment by roseduan** [comment]
there are some comments, please review @Reid00

**Comment by Reid00** [comment]
fixed those issues, please take a look.

**Pull request merged**